python/pyarrow/tests/test_fs.py

```diff
@@ -255,7 +255,10 @@ def s3fs(request, s3_server):
         allow_move_dir=False,
         allow_append_to_file=False,
     )
-    fs.delete_dir(bucket)
+    try:
+        fs.delete_dir(bucket)
+    except OSError:
+        pass
```

Comment:
Note: this is the "cheap" solution to #37692. Otherwise pretty much all tests using the `s3fs` fixture fail.

Reply:
I would like @jorisvandenbossche feedback on this one as I am not sure whether this fix is acceptable as it might hide other issues that could potentially happen.

Reply:
We should certainly fix the underlying issue instead of trying to hide it like this.

Reply:
I don't have time today to look myself in detail at it, but let's first try to investigate a bit more the cause of https://github.com/apache/arrow/issues/37692. 

Can we "just" skip the tests for now in the conda recipe tests if we want to move forward with this PR in the meantime?

Reply:
Yes, so what Antoine said ;)

Reply:
Yes we can skip the tests, but I'd argue that that's just a different flavour of "hiding it" 😋

Reply:
We know those tests are passing right now on various other CI builds; they are (probably) only failing here because of using a "wrong" minio version (of course we should ensure to make them work with newer minio as well, but there is a reason it is pinned at the moment). Hiding with a try/except would also hide potential future errors on all those other CI builds that are working right now.

Reply:
Fine by me. I'll remove minio and the try-except hack here.